mkat_fpga_tests/test_cbf.py

```diff
@@ -336,6 +339,124 @@ def print_baselines():
+
+    def test_back2back_consistency(self):
+        """1. Check that back-to-back dumps with same input are equal"""
+        test_name = '{}.{}'.format(strclass(self.__class__), self._testMethodName)
+        init_dsim_sources(self.dhost)
+        test_chan=1500
+
+        requested_test_freqs = self.corr_freqs.calc_freq_samples(
+            test_chan, samples_per_chan=9, chans_around=1)
+        expected_fc = self.corr_freqs.chan_freqs[test_chan]
+        self.dhost.sine_sources.sin_0.set(frequency=expected_fc, scale=0.25)
+
+        for i, freq in enumerate(requested_test_freqs):
+            print ('Testing dump consistancy {}/{} @ {} MHz.'.format(
+                i+1, len(requested_test_freqs), freq/1e6))
+            self.dhost.sine_sources.sin_0.set(frequency=freq, scale=0.125)
+            dumps_data = []
+            for dump_no in range(3):
+                if dump_no == 0:
+                    this_freq_dump = self.receiver.get_clean_dump(DUMP_TIMEOUT)
+                    init_data_dump = np.max(this_freq_dump['xeng_raw'])
+                else:
+                    this_freq_dump = self.receiver.data_queue.get(DUMP_TIMEOUT)
+                this_freq_data = this_freq_dump['xeng_raw']
+                dumps_data.append(this_freq_data)
+
+            diff_dumps = []
+            for comparison in range(1, len(dumps_data)):
+                d0 = dumps_data[comparison - 1]
+                d1 = dumps_data[comparison]
+                diff_dumps.append(np.max(d0 - d1))
+
+            dumps_comp = np.max(np.array(diff_dumps)/init_data_dump)
+            self.assertLess(dumps_comp, self.threshold,
+                'dump comparison ({}) is >= {} threshold[dB].'
+                    .format(dumps_comp, self.threshold))
+
+    def test_freq_scan_consistency(self):
+        """2. Check that identical frequency scans produce equal results"""
+        test_name = '{}.{}'.format(strclass(self.__class__), self._testMethodName)
+        init_dsim_sources(self.dhost)
+        test_chan=1500
+
+        requested_test_freqs = self.corr_freqs.calc_freq_samples(
+            test_chan, samples_per_chan=9, chans_around=1)
+        expected_fc = self.corr_freqs.chan_freqs[test_chan]
+        self.dhost.sine_sources.sin_0.set(frequency=expected_fc, scale=0.25)
+        init_dsim_sources(self.dhost)
+
+        scans = []
+        for scan_i in range(3):
+            scan_dumps = []
+            scans.append(scan_dumps)
+            for i, freq in enumerate(requested_test_freqs):
+                print ('{} of {}: Testing frequency scan consistancy {}/{} @ {} MHz.'.format(
+                scan_i+1, len(range(3)), i+1, len(requested_test_freqs), freq/1e6))
+                if scan_i == 0:
+                    self.dhost.sine_sources.sin_0.set(frequency=freq, scale=0.125)
+                    this_freq_dump = self.receiver.get_clean_dump(DUMP_TIMEOUT)
+                    init_freq_dump = np.max(this_freq_dump['xeng_raw'])
```

Comment:
Perhaps confusing variable name? Perhaps initial_freq_max? Also, you need to save this value for each frequency (at the moment you are overwriting it each time the frequency is changed). You should perhaps store each frequency's max in a list and then normalise each frequency separately in the comparison loop. IOW, loop over the dumps and then over the frequencies and get the correct initial_freq_max value for that frequency from the list.

Similar comments for the restart-test.